src/smk/layer/layer-vector.js

```diff
@@ -24,7 +24,8 @@ include.module( 'layer.layer-vector-js', [ 'layer.layer-js' ], function () {
         const legendData = [];
         legendData.push({
             title: self.config.legend && self.config.legend.title || self.config.title,
-            style: self.config.style
+            style: self.config.style,
+            hasConditionalStyling: self.config.conditionalStyles || false
```

Comment:
While JavaScript allows the assignment of an object or a boolean to the same variable, it is not good practice. I'd suggest assigning true/false or assigning the object/undefined to `hasConditionalStyling`.

Reply:
I'm checking for the existence of the "conditionalStyles" property in config, but yes, it is kind of dumb to actually assign it when it's present :) . I'll update this to do a better check of existence.